internal/compiler/generator/cpp.rs

```diff
@@ -1852,6 +1861,29 @@ fn generate_sub_component(
         ));
     };
 
+    let mut item_geometry_cases = vec!["switch (index) {".to_string()];
+    item_geometry_cases.extend(
+        component
+            .geometries
+            .iter()
+            .enumerate()
+            .filter_map(|(i, x)| x.as_ref().map(|x| (i, x)))
```

Comment:
Just curious, isn't `x` in `x.as_ref()` already a reference to an expression since the iterator was created using `into()` instead of `into_iter()`?

Reply:
x is of type `&Option<MutExpression>`,  and i need to change that to `Option<&MutExpression>`